### PR TITLE
Remove Tiled Mosaic UI and default override

### LINKED ISSUE #1287: Tiled Galleries: column option not available for Thumbnail Grid Gallery types.
In 1983051-t it was reported that the option to choose the number of columns in a Thumbnail Grid gallery was not there. 

Since column options are not available with Tiled Mosaic galleries, if the Tiled Galleries module in Jetpack is active, the ability to choose the column number for all galleries is automatically not displayed. In order for the user to be able to toggle the number of columns, they need to go to Settings > Media, and deselect "Display all your gallery pictures in a cool mosaic." Having that option "on" overwrites how gallery types work:

https://github.com/Automattic/jetpack/blob/3.2/modules/tiled-gallery/tiled-gallery.php#L163

This doesn't allow for a check of what type of gallery is being selected (and subsequently prevents the display of gallery type-specific options):

https://github.com/Automattic/jetpack/blob/3.2/_inc/gallery-settings.js#L19

I was able to reproduce in my Sandbox. This is the view when "Display all your gallery pictures in a cool mosaic." is selected on Settings > Media:

no_column_options

Deselecting that option in Settings > Media allows the columns option to reappear: 

columns_option_returns

We should allow the column option appear by default when Thumbnail Grid is the selected gallery type, even if the Tiled Mosaic option is selected in Settings > Media.

### PR DESCRIPTION
**Pull request by aheckler**
Removes the Tiled Mosaic checkbox from Settings > Media and stops Jetpack from overriding the default gallery type with Tiled Mosaic. IMHO, this PR simplifies gallery behavior with Jetpack given that:

- Previously, users had to check an obscure box in Settings > Media before TM's would work. With this PR, the checkbox isn't needed. TM's can always be selected as a gallery type if the module is active.
- Previously, Jetpack was overriding the default gallery type (Thumbnail Grid) with TM when the above checkbox was enabled, potentially confusing gallery shortcode users. This PR stops that behavior.

In short, by merging this PR, things are much more intuitive: with the TM module active, Jetpack's three new gallery types become available in the gallery shortcode and Add Media modal. No more, no less.

Fixes #1287.

### CODE REVIEW

**Comment by jeherve** [comment]
> Previously, Jetpack was overriding the default gallery type (Thumbnail Grid) with TM when the above checkbox was enabled, potentially confusing gallery shortcode users. This PR stops that behavior.

I liked that behaviour. It allowed me to use Tiled Galleries on all galleries of my site, without having to go back and edit the dozens of galleries I created before Jetpack was installed on my site.

**Comment by aheckler** [comment]
> I liked that behaviour. It allowed me to use Tiled Galleries on all galleries of my site, without having to go back and edit the dozens of galleries I created before Jetpack was installed on my site.

Maybe we could add a filter that would do this for the sake of backward compatibility? In fact, why not allow the user to set *any* gallery type as the default?

From a new user's point of view, I still think the way TM works now is counter-intuitive. I feel like, if I were brand new to Jetpack and enabled Tiled Galleries, I would reach a stumbling point when I enabled the module but then saw no new gallery types appear in the Add Media modal. I can check the box under Settings > Media to make them appear, but then TM becomes the default, which I may not want.

**Comment by jeherve** [comment]
> why not allow the user to set any gallery type as the default?

I like that.

> I would reach a stumbling point when I enabled the module but then saw no new gallery types appear in the Add Media modal.

I don't understand. If the new gallery types don't appear until you check the box under Settings > Media, that's a bug.

**Comment by aheckler** [comment]
> I don't understand. If the new gallery types don't appear until you check the box under Settings > Media, that's a bug.

I will do some double-checking today to see if this is the case.

**Comment by aheckler** [comment]
1. Install WP 4.1.1 + JP 3.3.2 on a brand new site.
2. Enable Jetpack and connect to WP.com.
3. Enable Tiled Galleries module.
4. The checkbox under Settings > Media is not checked.
5. Upload some media.
6. Create a new post.
7. Insert a gallery. Jetpack's gallery types are selectable.

I could swear that they were *not* selectable when I tested this the other day on my sandbox. I tried testing just now but I think I broke my sandbox in some way (the Add Media modal doesn't work at all).

**Comment by aheckler** [comment]
> I could swear that they were not selectable when I tested this the other day on my sandbox.

I must have been wrong (or something was corrupted somewhere): after a WP reinstall on my sandbox, new gallery options show up once TG module is activated. No need to check the box.

Will need to look into how to add the filter for setting the default gallery type to any type.

**Comment by samhotchkiss** [comment]
> > why not allow the user to set any gallery type as the default?
> I like that.

So many settings!

**Comment by aheckler** [comment]
@samhotchkiss Which is better in your opinion:

A) The way Tiled Galleries work now. That is, you can enable the module to use the new gallery types, and also check a box in Settings > Media to make TG's the default if you want.

or...

B) A hypothetical future state, where instead of just being able to make TG's the default, you could make any gallery type the default. (Basically just replacing the existing setting with a more comprehensive one.)

I like B better but this PR is getting unwieldy, so maybe I can start a new one for that if it'd help.

**Comment by samhotchkiss** [comment]
How about C-- when you turn on Jetpack, Tiled Mosaics become the default for any *new* gallery?  There's no setting, it just works.  Default can be changed via filter.

Death to settings!

**Comment by aheckler** [comment]
> when you turn on Jetpack, Tiled Mosaics become the default for any new gallery?

I feel like that wouldn't be a good user experience. :confused: I'd only expect that the new gallery types would become available, not that they'd take over the defaults.

IMHO, ideally what should happen is the Settings > Media checkbox goes away, and instead we add a filter that allows you to set any gallery type as the default. If you don't use the filter, we don't override the default (Thumbnail Grid).

**Comment by kraftbj** [comment]
@aheckler I think you were right about not seeing the option before. I stumbled upon it last night too in #1681 .

**Comment by jeherve** [comment]
> B) A hypothetical future state, where instead of just being able to make TG's the default, you could make any gallery type the default. (Basically just replacing the existing setting with a more comprehensive one.)

I like that. It wouldn't be a new option, just replacing a checkbox by a dropdown.

**Comment by aheckler** [comment]
Going to close this since it's more complicated than necessary and against the general consensus here.
